packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,125 @@
+import { sendMessageToParent } from '../internal/communication';
+import { registerHandler, removeHandler } from '../internal/handlers';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Allows your application to interact with the host M365 application's search box.
+ * By integrating your application with the host's search box, users can search
+ * your app using the same search box they use elsewhere in Teams, Outlook, or Office.
+ *
+ * This functionality is in Beta.
+ * @beta
+ */
+export namespace search {
+  const onChangeHandlerName = 'search.queryChange';
+  const onClosedHandlerName = 'search.queryClose';
+  const onExecutedHandlerName = 'search.queryExecute';
+
+  /**
+   * This interface contains information pertaining to the contents of the host M365 application's search box
+   */
+  export interface SearchQuery {
+    /** The current search term in the host search experience */
+    searchTerm: string;
+
+    /**
+     * Timestamp sequence value to ensure messages are processed in correct order / combine them.
+     * Having any sort of logic around combining messages or sorting them
```

Comment:
I would remove the part of the comment on line 29/30. That was more an idea for you to consider. It's okay with me (at least until you get developer feedback) to leave it to app developers to handle combing/sorting messages